plugins/woocommerce-blocks/assets/js/base/components/drawer/index.tsx

```diff
@@ -56,11 +56,12 @@ const CloseButtonPortal = ( {
 		? createPortal(
 				<Button
 					className="wc-block-components-drawer__close"
-					icon={ close }
 					onClick={ onClick }
-					label={ __( 'Close', 'woocommerce' ) }
-					showTooltip={ false }
-				/>,
+					removeTextWrap
+					aria-label={ __( 'Close', 'woocommerce' ) }
+				>
+					<Icon icon={ close } />
```

Comment:
I have opted not to support the `icon` arg. But of course if we think we should retain it for backwards compat I'm open to it.

These APIs are definitely bad though, most things you can do by rendering a child instead of passing a prop and its almost always much more flexible, reusable etc.

Reply:
Do you think we should keep it in, but make it show a deprecation notice along with instructions on the better solution? Would it be too complex to support both approaches at the same time?

Reply:
Not too complex. Happy to add deprecation instead :)

Reply:
@opr I just looked a little bit deeper into this, and I just want to maybe challenge the idea to support `icon` here and see what you think.

The reason `icon` works is that it is part of the underlying `Button` API of the wordpress component. So we were just passing it through along with any other prop that WP button supports. We used this internally, but we never documented it in Storybook and I don't see any documentation that implies you can just use any prop supported by WP button.. well that is except for the TS type we use for the props, but that was fairly recent and more a result of converting to TS.

If we continue to support `icon` I think it implies we'd support the rest of the icon api supported on WP button as well, such as `iconPosition` and `iconSize`. We'd also be supporting it only for anyone who externally consumes this component, because as you can see here it was trivial for me to just render an icon in the children, so internally we don't need this.

I think we should just ditch support for passing through any of the old props of WP button because I think you could argue if we don't support all of them with deprecation warnings we shouldn't support any of them.

I would prefer to just support the obviously exposed props such as `showSpinner` (which I have). Wdyt?

I also think we could/should deprecate `showSpinner` because its another situation where composing components is better, but I won't suggest it here because unless someone gets given the task of working on f/e components framework I think that deprecation warning will stick around a long time.

Reply:
Thanks for the thoughts here, it makes sense to discontinue support for the WP button props, and I think it falls outside of our usual way of deprecating props (given they were never advertised). That said, we should make sure we announce this ahead of the release that includes this change, as part of a developer advisory. In fact, we could even do that now in a discussion. What do you think?

I agree `showSpinner` is a little more complicated to remove because we have advertised it in documentation (Storybook) but you're right, composition makes more sense. 

>  I won't suggest it here because unless someone gets given the task of working on f/e components framework I think that deprecation warning will stick around a long time.

I don't know if the deprecation notice being around for a long time is really a problem. Arguably, the longer it's around for the better. It gives folks more time to move away from it, maybe we _should_ add it as part of this effort?

Reply:
> I don't know if the deprecation notice being around for a long time is really a problem. Arguably, the longer it's around for the better. It gives folks more time to move away from it, maybe we should add it as part of this effort?

Yep I can get behind that argument. I'll add a deprecation here.

Reply:
Deprecation resolved: https://github.com/woocommerce/woocommerce/pull/45974/commits/f9f4d43aae58e7ce849ef5707a60b2ecdc276907

Reply:
> If we continue to support icon I think it implies we'd support the rest of the icon api supported on WP button as well, such as iconPosition and iconSize.

If there are users of this WP icon API via the Button component, what options would they have in the future to achieve equivalent output?

As the surface area is big, and extensions might be using this button in their implementations, I am a bit concerned with deprecating this `icon` prop without sharing ways [code examples] to achieve the same behavior.

Ideally, we should map the current API for the 1st phase and document improvements for future work on the components. 

These deprecations could be announced ahead of time with a communicated removal in a specific WooCommerce version.

Reply:
> Ideally, we should map the current API for the 1st phase and document improvements for future work on the components.

It's just that this usage was never documented or suggested in Storybook. So what could be considered the "API". props like `icon` and `iconPosition` and such are imo API of WP button but not API of this button. It's murky though I know. I can understand your concern.

> If there are users of this WP icon API via the Button component, what options would they have in the future to achieve equivalent output?

>As the surface area is big, and extensions might be using this button in their implementations, I am a bit concerned with deprecating this icon prop without sharing ways [code examples] to achieve the same behavior.

That part is pretty easy, rendering an icon as a child instead like I have done here in this LOC